src/main/java/com/sucy/skill/dynamic/condition/ValueTextCondition.java

```diff
@@ -0,0 +1,55 @@
+package com.sucy.skill.dynamic.condition;
+
+import com.sucy.skill.dynamic.DynamicSkill;
+import org.bukkit.entity.LivingEntity;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiPredicate;
+import java.util.regex.Pattern;
+
+/**
+ * A condition for dynamic skills that requires the text value to match the settings
+ */
+public class ValueTextCondition extends ConditionComponent {
+
+    private static final String MODE   = "mode";
+    private static final String VALUE  = "value";
+    private static final String EXPECT = "expect";
+
+    @Override
+    public String getKey() {
+        return "text";
+    }
+
+    @Override
+    public boolean execute(LivingEntity caster, int level, List<LivingEntity> targets, boolean force) {
+        return test(caster, level, null) && executeChildren(caster, level, targets, force);
+    }
+
+    @Override
+    boolean test(final LivingEntity caster, final int level, final LivingEntity target) {
+        final CompareMode mode = CompareMode.valueOf(settings.getString(MODE).toUpperCase());
+        final Object value = DynamicSkill.getCastData(caster).get(VALUE);
```

Comment:
This should probably be grabbing the value from the skill configuration rather than passing the static string in.